rust/lance-core/src/format/manifest.rs

```diff
@@ -193,15 +200,56 @@ impl WriterVersion {
         let tag = parts.next();
         Some((major, minor, patch, tag))
     }
+
+    pub fn semver_or_panic(&self) -> (u32, u32, u32, Option<&str>) {
```

Comment:
any reason we don't want to use the feature flag system here?

Reply:
What's that?

Reply:
Or, more accurately, how does that apply?

Reply:
Ah, I think I see what you mean.  I was thinking "rust features" but I think you mean the lance writer features.

Instead of relying on the writer version we could look for `writer_feature_flags`.  I'm not sure.  Technically, there isn't a new proto feature here.  We didn't make any changes to how the fragment bitmap is stored in the proto.  The problem is specific to our implementation of Lance.  E.g. if some other writer was creating lance files then they should be fine.

It also feels a little weird to have a writer feature named "no really, you can trust the fragment bitmap now"

CC @wjones127 for second opinion

Reply:
What Weston saying is exactly what I was thinking. It's possible in the future we will have multiple reader / writer implementations, and I want those flags to be meaningful to them.